src/datta_lab_to_nwb/markowitz_gillis_nature_2023/fiberphotometryinterface.py

```diff
@@ -0,0 +1,198 @@
+        metadata_schema["properties"]["FiberPhotometry"] = {
+            "type": "object",
+            "properties": {
+                "area": {"type": "string"},
+                "reference_max": {"type": "number"},
+                "signal_max": {"type": "number"},
+                "signal_reference_corr": {"type": "number"},
+                "snr": {"type": "number"},
+            },
+        }
+        return metadata_schema
+
+    def run_conversion(self, nwbfile: NWBFile, metadata: dict):
+        """Run conversion of data from the source file into the nwbfile."""
+        session_df = pd.read_parquet(
+            self.source_data["file_path"],
+            columns=self.source_data["columns"],
+            filters=[("uuid", "==", self.source_data["session_uuid"])],
+        )
+
+        # Excitation Sources Table
+        excitation_sources_table = ExcitationSourcesTable(
+            description="ExcitationSourcesTable",
+        )
+        excitation_sources_table.add_row(
+            peak_wavelength=480.0,
+            source_type="laser",
+        )
+        excitation_sources_table.add_row(
+            peak_wavelength=400.0,
+            source_type="laser",
+        )
+
+        # Photodetectors Table
+        photodetectors_table = PhotodetectorsTable(
+            description="PhotodetectorsTable",
+        )
+        photodetectors_table.add_row(peak_wavelength=527.0, type="PMT", gain=1.0)
+
+        # Fluorophores Table
+        fluorophores_table = FluorophoresTable(description="fluorophores")
+
+        fluorophores_table.add_row(
+            label="dlight1.1",
+            location=metadata["FiberPhotometry"]["area"],
+            coordinates=(0.260, 2.550, -2.40),  # (AP, ML, DV)
+        )
+
+        # Fibers Table
+        fibers_table = FibersTable(
+            description="FibersTable",
+        )
+        nwbfile.add_lab_meta_data(
+            FiberPhotometry(
+                fibers=fibers_table,
+                excitation_sources=excitation_sources_table,
+                photodetectors=photodetectors_table,
+                fluorophores=fluorophores_table,
+            )
+        )
+
+        # Important: we add the fibers to the fibers table _after_ adding the metadata
+        # This ensures that we can find this data in their tables of origin
+        fibers_table.add_fiber(
+            excitation_source=0,  # integers indicated rows of excitation sources table
+            photodetector=0,
+            fluorophores=[0],  # potentially multiple fluorophores, so list of indices
+            location=metadata["FiberPhotometry"]["area"],
+            notes="None",
+        )
+        fibers_table.add_fiber(
+            excitation_source=1,  # integers indicated rows of excitation sources table
+            photodetector=0,
+            fluorophores=[0],  # potentially multiple fluorophores, so list of indices
+            location=metadata["FiberPhotometry"]["area"],
+            notes="None",
```

Comment:
Huh, is `notes` really a required property?

Reply:
Yeah, fibers_table in ndx_photometry needs it to be specified.

Reply:
I've requested that it be made optional https://github.com/catalystneuro/ndx-photometry/issues/5

Leaving this here as a note to try to remember to come back and remove it later